### DISCUSSION TITLE
mpbas error help

### DISCUSSION THREAD

**Discussion by cravenhydros** [modflowpy/flopy#2055] [Q&A] [answered]
Hello,

I am working on adding MODPATH 7 to an existing workflow. I am using MODFLOW-NWT.

I am using to load the model.
mod= flopy.modflow.Modflow.load()
>>> mod
MODFLOW 8 layer(s) 145 row(s) 130 column(s) 30 stress period(s)

mp = flopy.modpath.Modpath7(modelname=mpnamb, flowmodel=mod, exe_name=exe_name, model_ws=ws)
>>> mp
MODPATH 7 model

mpbas = flopy.modpath.Modpath7Bas(mp, porosity=0.1)

mpbas is giving me this error:

>>> mpbas = flopy.modpath.Modpath7Bas(mp, porosity=0.1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Anaconda\lib\site-packages\flopy\modpath\mp7bas.py", line 67, in __init__
    self.laytyp = Util2d(self.parent, (shape[0],), np.int32,
  File "C:\Anaconda\lib\site-packages\flopy\utils\util_array.py", line 1728, in __init__
    self.parse_value(value)
  File "C:\Anaconda\lib\site-packages\flopy\utils\util_array.py", line 2621, in parse_value
    raise Exception('Util2d:unsupported type in util_array: ' +
Exception: Util2d:unsupported type in util_array: <class 'NoneType'>

Am I missing something in the steps? Any ideas why Util2d is complaining?

Thanks!

**Comment by wpbonelli** [comment]
@cravenhydros the MP7 bas class tries to load `laytyp` from the flow model, but it looks like it's `None` in your trace, which could occur if NWT has no UPW (upstream weighting) package. Maybe something to double check?

Happy to take a look if you can upload input files or share the full model setup code, if the above isn't the issue.

**Reply by cravenhydros** [reply]
Hello, sorry I'm not able to share model files. 

when I look at 
>> mod.upw  
I get the following output to the screen.

hani = <flopy.utils.util_array.Util3d object at 0x0000021F6A385CA0> ('flopy.utils.util_array.Util3d)
 hdry = -1e+30 ('float)
 hk = <flopy.utils.util_array.Util3d object at 0x0000021F6A385C70> ('flopy.utils.util_array.Util3d)
 ipakcb = 50 ('int)
 iphdry = 1 ('int)
 layavg = <flopy.utils.util_array.Util2d object at 0x0000021F6A385AF0> ('flopy.utils.util_array.Util2d)
 laytyp = <flopy.utils.util_array.Util2d object at 0x0000021F6A385AC0> ('flopy.utils.util_array.Util2d)
 layvka = <flopy.utils.util_array.Util2d object at 0x0000021F6A385BB0> ('flopy.utils.util_array.Util2d)
 laywet = <flopy.utils.util_array.Util2d object at 0x0000021F6A385C10> ('flopy.utils.util_array.Util2d)
 npupw = 0 ('int)
 options =   ('str)
 ss = <flopy.utils.util_array.Util3d object at 0x0000021F6A385D00> ('flopy.utils.util_array.Util3d)
 sy = <flopy.utils.util_array.Util3d object at 0x0000021F6A385D30> ('flopy.utils.util_array.Util3d)
 unit_number = 11
 vka = <flopy.utils.util_array.Util3d object at 0x0000021F6A385CD0> ('flopy.utils.util_array.Util3d)
 vkcb = <flopy.utils.util_array.Util3d object at 0x0000021F6A385D60> ('flopy.utils.util_array.Util3d)

**Reply by wpbonelli** [reply]
Tricky to diagnose without more detail &mdash; any way to trim the model down to something that can be shared and minimally reproduces the issue?

In the meantime you could check `laytyp` is not None on either `mod` (mfnwt model) or `mp` (mp7 model)

**Reply by cravenhydros** [reply, marked as answer]
I updated FP from flopy 3.2.11 to the most recent release and the issue seems to be resolved.

**Reply by wpbonelli** [reply]
Glad to hear
